### DISCUSSION TITLE
Drift Longevity

### DISCUSSION THREAD

**Discussion by IttaiBarkai** [simolus3/drift#3252] [General]
Hi there,

My team and I are thinking about migrating from Isar to Drift due to some concerns around the lack of communication from the main developer, which the package is highly dependent on. 

Before making the switch to Drift, we just wanted to check if there are any measures in place to ensure the longevity of this package so we don't run into the same issues down the road?

Thanks!

**Comment by dickermoshe** [comment]
I hear you. I was using Isar too and got burned.

But I think there is a big difference between Isar and Drift

1. Isar is written in rust and dart. This really limits the pool of potential contributors. Drift is just good ol plain dart code.
2. Drift has been actively maintained by @simolus3 for almost 6 now. At no point has there been a pause in development as you can see from this commit graph.

Isar: 
<img width="399" alt="image" src="https://github.com/user-attachments/assets/5c04543a-bd7c-41f7-8827-df8279273453">

Drift: 
<img width="397" alt="image" src="https://github.com/user-attachments/assets/1b202aac-4b0f-4d20-bfb5-595ad7b28744">

3. Drift has more contributors and collaborates than Isar. 

Ultimately, no-one can predict the future, but I would bet on drift.

**Reply by IttaiBarkai** [reply]
Thank you for the quick response and insight. I have to admit that it's a bit concerning that this package is also essentially maintained by a single person. However, the steady development and active maintenance in comparison to Isar are reassuring.

**Reply by dickermoshe** [reply]
It's 2 people

**Reply by simolus3** [reply]
> Before making the switch to Drift, we just wanted to check if there are any measures in place to ensure the longevity of this package so we don't run into the same issues down the road?

What kind of measures do you have in mind here? I think drift is in a healthy position right now (it's not just me anymore thanks to @dickermoshe doing invaluable work on new features and maintenance). But even before that, this project benefited from active community members contributing or answering questions. And I'm of course also still here and actively contributing, almost always responding to new issues within a day.

Another thing is that drift is (I'd say) a pretty stable project (apart from new features, major work mostly resolves around supporting new ecosystem features like dart2wasm and new FFI features / macros coming up). So both from a technical as well as from an organizational standpoint, we're doing okay.

**Reply by IttaiBarkai** [reply]
Thanks for the replies @dickermoshe and @simolus3 . Knowing that there are actually two developers actively maintaining and contributing to Drift is already a big confidence boost (apologies for missing that), as well as hearing directly from you guys adds even more reassurance. 

An issue we had with Isar was that having only one developer with the authority to merge code meant even small dependency updates or community contributions were delayed or never merged when they were unavailable. I’m glad to hear that’s not the case with Drift. 

Thanks for taking the time to respond.
